x-pack/plugins/ml/public/application/explorer/anomaly_timeline_state_service.ts

```diff
@@ -185,6 +187,12 @@ export class AnomalyTimelineStateService extends StateService {
       );
       this._viewBySwimlaneFieldName$.next(viewBySwimlaneFieldName);
       this._viewBySwimLaneOptions$.next(viewBySwimlaneOptions);
+
+      if (selectedCells && selectedCells.viewByFieldName !== undefined) {
+        this._selectedInfluencers$.next(
+          getSelectionInfluencers(selectedCells, selectedCells.viewByFieldName)
+        );
+      }
```

Comment:
I see 2 issues here:
1. `_selectedInfluencers` has nothing to do with this subscription. It only depends on `selectedCells`. Hence, it should listen for `this._selectedCells$` and derive the value from it. You can achieve it without creating a new subject, but simply by creating a new observable with a pipe. Check `getSwimLaneBucketInterval$()` for inspiration :)
2. The is a bug, actually. If the user selects cells on the Overall swim lane, the service still tells that there is influencer selection.

Reply:
Thinking about it some more, I'm not sure if there's a clear benefit of making this an observable looking through current usage of `getSelectionInfluencers`. Also, refactoring all current usages of getSelectionInfluencers to take advantage the observable also seems out of the scope of this fix PR 🤔 I think I'll revert this change and add a TODO comment for future improvements.

Reply:
As discussed in Slack, I've reverted this addition of `getSelectionInfluencers$` inside AnomalyTimelineStateService and added a TODO comment to refactor usage of  `getSelectionInfluencers`, `getSelectionJobIds`, and `getSelectionTimeRange` in a future follow-up here [`8551773` (#140147)] (https://github.com/elastic/kibana/pull/140147/commits/85517732200916a237c0eba9f0b7651a7a1ee52e)